glyphs-reader/src/font.rs

```diff
@@ -179,9 +179,58 @@ pub struct Glyph {
 #[derive(Debug, PartialEq, Hash)]
 pub struct Layer {
     pub layer_id: String,
+    pub associated_master_id: Option<String>,
     pub width: OrderedFloat<f64>,
     pub shapes: Vec<Shape>,
     pub anchors: Vec<Anchor>,
+    pub attributes: LayerAttributes,
+}
+
+impl Layer {
+    pub fn is_master(&self) -> bool {
+        self.associated_master_id.is_none()
+    }
+
+    pub fn is_intermediate(&self) -> bool {
+        self.associated_master_id.is_some() && self.attributes.coordinates.is_some()
+    }
+
+    // TODO add is_alternate, is_color, etc.
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Hash)]
+pub struct LayerAttributes {
+    pub coordinates: Option<Vec<OrderedFloat<f64>>>,
```

Comment:
Unless an empty Vec has different meaning than None, which is rare and weird, use Vec rather than Option<Vec>

Reply:
good point, they ought to mean the same thing. An empty "coordinates" vector makes no sense; if it did, we could risk treating that not-quite-intermediate layer as a duplicate location (as it would overlap with its own associated master location). I'll remove the Option.